heartbeat/monitors/resolver.go

```diff
@@ -0,0 +1,99 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package monitors
+
+import (
+	"fmt"
+	"net"
+)
+
+// Resolver lets us define custom DNS resolvers similar to what the go stdlib provides, but
+// potentially with custom functionality
+type Resolver interface {
+	// ResolveIPAddr is an analog of net.ResolveIPAddr
+	ResolveIPAddr(network string, host string) (*net.IPAddr, error)
+	// LookupIP is an analog of net.LookupIP
+	LookupIP(host string) ([]net.IP, error)
+}
+
+// stdResolver uses the go std library to perform DNS resolution.
+type stdResolver struct{}
```

Comment:
wdyt about adding unit tests for this one?

Reply:
I think you're right to add tests for the static resolver which I've done, but given that this one just passes the functions through to stdlib, I don't see how we could write significantly useful tests. We could compare output for a limited set of inputs but that doesn't say much. IMHO, the utility of those tests is pretty low.

It's not a huge deal, I'll add some simple ones if you think it's important, but I don't think tests on `stdResolver` will decrease chances of a future regression.